xradar/accessors.py

```diff
@@ -204,3 +218,28 @@ def add_crs(self) -> dt.DataTree:
         """
         ds = self.xarray_obj
         return ds.pipe(add_crs_tree)
+
+    def apply(self, func, *args, **kwargs):
+        """
+        Applies a given function to all sweeps in the radar volume.
+
+        Parameters
+        ----------
+        func : function
+            The function to apply to each sweep.
+        *args : tuple
+            Additional positional arguments to pass to the function.
+        **kwargs : dict
+            Additional keyword arguments to pass to the function.
+
+        Returns
+        -------
+        DataTree
+            The DataTree object after applying the function to all sweeps.
+        """
+        radar = self.xarray_obj
+        for key in list(radar.children):
+            if "sweep" in key:
+                radar[key].ds = func(radar[key].to_dataset(), *args, **kwargs)
```

Comment:
This will overwrite the current tree nodes, as stated in the doc string. Is this really what we want, @syedhamidali? We also could think to construct a new tree here, keeping the original tree as is.

In `georeference()` we also overwrite most likely as we just add the georeferencing x,y,z coordinates. But for the sake of compatibility/comparability of API's it would make sense to return a new DataTree here.

I'd like to get some more opinions, @openradar/xradar WDYT?

Reply:
I agree with @kmuehlbauer. Could we use an `inplace` argument in case users want to overwrite?

I would like recommend this improved version for looping through the `datatree` object. This might be more `datatree` compatible...

```python
for node in radar.match("sweep*").subtree:
    radar[node.path].ds = func(radar[node.path].to_dataset(), *args, **kwargs)
```

Reply:
@aladinor Your `datatree`-knowledge is really impressive. This looks very neat!

How would your implementation look like to create a new datatree instead of overwriting the sweep-nodes?

Reply:
> Perhaps, we could use an `inplace` argument if users want to overwrite

IIRC, xarray remove all those `inplace`-implementations from their code base and only return. So maybe we just stick to their behaviour?

Reply:
I was thinking in something like this, 

```python


radar = self.radar
tree = {}
for node in radar.match("sweep*").subtree:
    tree[node.path] = func(radar[node.path].to_dataset(), *args, **kwargs)
return DataTree.from_dict(tree)
```

However, we need to add the other radar_parameters, georeferencing... nodes, as well as the `DataArray`s in the root before returning it.

Reply:
Or copying it and replacing the nodes we want to modify(?)

```python

radar = self.radar.copy()

for node in radar.match("sweep*").subtree:
    radar[node.path] = func(radar[node.path].to_dataset(), *args, **kwargs)
return radar
```

Reply:
I like the `.from_dict()`-approach as it is also used very much in `datatree` itself. Let's try to figure out a solution going that way.

Reply:
How about:

```python
# root Dataset
tree = {"/": radar.ds}
# all but root dataset
tree.update({node.path: node for node in radar.subtree if node.path != "/"})
# overwrite sweep nodes
tree.update({node.path: func(radar[node.path].to_dataset(), *args, **kwargs) for node in radar.match("sweep*").subtree if node.path != "/"})
# return new Datatree
return datatree.DataTree.from_dict(tree)
```

Reply:
It looks great! Should we start using `datatree` directly from xarray ?

```python 
from xarray.core.datatree import DataTree
...
```

Reply:
@aladinor

It is not officially merged yet. So maybe we should keep importing from datatree until this is officially announced. 

You have been attending those meetings, so I'd rely on your expertise if it is safe to move fast here.